### DISCUSSION TITLE
Loopback Serverless (Large Scale)

### DISCUSSION THREAD

**Discussion by ghost** [loopbackio/loopback-next#7522] [Q&A] [answered]
@raymondfeng did an amazing example serverless (added a link to it below). I wanted to get an idea of using this concept in a larger context. 

Let’s say I wanted to incrementally move parts of server logic (a larger lb4 application) into serverless functions. Would I have many versions of what @raymondfeng did in the example for each function? Or did @raymondfeng imagine having multiple functions in this one repository that is then deployed. We can think of a function as replacing a REST api to keep things clear. 

Note: 

After reading a bit more, I am seeing that a loopback application is being deployed as a serverless function. I think my questions still remain to how to do this on a larger scale and how an application is broken up. 
 
Thank you all!
 
Sirrele Steinfeld
 
 
Repo: https://github.com/raymondfeng/loopback4-example-serverless

**Comment by raymondfeng** [comment, marked as answer]
You can possibly have a controller that injects an extensible list of services, each of which represents a serverless function. The controller hooks with the serverless container andvit delegates to registered services based on the incoming request.

**Reply by ghost** [reply]
Awesome and makes sense! This was helpful and I will give this a try. Thank you for your quick responses
